### ISSUE TITLE
[feature] need a process undefined linting rule

### ISSUE THREAD

**Issue by wraithgar** [npm/eslint-config#70]
Code that checks for the presence of process should test its type against undefined. Trying to access it AT ALL in a browser environment will throw an exception.

I don't know if a linting rule exists or if one needs to be written.  But it should prevent things like:

```javascript
if (process) {
```

or

```javascript
if (process?.foo) {
```

**Comment by wraithgar** [comment]
`require('process')` would make more issues in places where we currently work correctly I think, since bundlers would start needing to either shim or throw.

**Comment by nlf** [comment]
yeah, that's the issue with that approach for sure. adding a comment to disable the no-undef rule on the lines that you actually use `process` would work and avoids that problem, but is also kind of wonky feeling

**Comment by wraithgar** [comment]
Yeah it still wouldn't prevent `process?.x` which is the real thing we need to prevent. If it doesn't prevent that it's all just pretend anyways.
